src/ray/observability/ray_event_interface.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2025 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+
+#include "src/ray/protobuf/public/events_base_event.pb.h"
+
+namespace ray {
+namespace observability {
+
+class RayEventInterface {
+ public:
+  virtual ~RayEventInterface() = default;
+
+  // Resource ID is a concept in Ray Event framework that captures the unique identifier
+  // of the resource that the event is associated with. For example, the source ID of
+  // a task is the pair of task ID and task attempt ID, for a driver job, it is the
+  // driver job ID.
+  //
+  // Resource ID is used for two purposes:
+  // 1. To associate the execution event with the definition event.
+  // 2. To merge the individual execution events into a single execution event (single
+  // data point to a time series).
+  virtual std::string GetResourceId() const = 0;
```

Comment:
is the string hex or binary of the ID? Also should we call it EntityId to avoid confusion with Ray resources.